devito/operations/interpolators.py

```diff
@@ -237,9 +266,39 @@ def _interp_idx(self, variables, implicit_dims=None, pos_only=()):
 
         # Substitution mapper for variables
         mapper = self._rdim._getters
-        idx_subs = {v: v.subs({k: c + p
-                    for ((k, c), p) in zip(mapper.items(), pos)})
-                    for v in variables}
+
+        cdmapper = {}
+        # Need to adjust bounds if Function defined on a SubDomain
+        if subdomain:
```

Comment:
my feeling is that most of these subdomains-related pieces of code should be moved into separate utility functions inside `types/subdomain.py`